### DISCUSSION TITLE
How to run with specific language?

### DISCUSSION THREAD

**Discussion by libook** [flameshot-org/flameshot#3279] [Q&A] [answered]
This tool is the most great one I have seen.
I would like to use it as a default screenshot tool in Gnome DE.
So I have tried to set a customized hotkey in the settings of Gnome.
I set the command as `env LANG=zh_CN.UTF-8 flameshot gui` but the GUI was still English.
I don't know much about C++ or Qt. It seems that the QLocale Class used environment variables like LANG, LC_ALL. But they didn't work.

If anyone can offer me some advice, I would greatly appreciate it.

**Comment by mmahmoudian** [comment]
What is your system language? How have you installed Flameshot? What do you get when you run `locale` in a new terminal?

**Reply by libook** [reply]
```bash
$ locale
LANG=zh_CN.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=en_US.UTF-8
```
The language of my Gnome DE is  zh_CN.

**Reply by mmahmoudian** [reply]
I think LC_ALL is what is used to determine the language.

**Reply by libook** [reply]
I have tried `env LC_ALL=zh_CN.UTF-8 flameshot gui` and it didn't work ether.

**Reply by mmahmoudian** [reply]
@veracioux I'm not sure but I think we are using https://doc.qt.io/qt-6/qlocale.html#setDefault to detect language, but I cannot file what exactly does Qt uses as the reference from the host machine. Would you please point us to the correct direction.

**Reply by veracioux** [reply]
@libook  @mmahmoudian  Both `LANG` and `LC_*` variables can be used to set the locale for Qt. There is a bug in flameshot, where localizations are only applied when you run the flameshot daemon (the command `flameshot` without arguments). I have a fix and will create a PR for that shortly.

**Reply by veracioux** [reply, marked as answer]
@libook You can watch https://github.com/flameshot-org/flameshot/pull/3313 which adds a fix for that. You can wait for it to be merged to master and then manually build flameshot (or if you are using an Arch Linux based OS, you can install `flameshot-git` from the AUR which is always up-to-date with master), or you can even build from the PR branch directly if you want. If you need help with any of this, I'll be happy to help.
